rskj-core/src/test/java/co/rsk/peg/BridgeSupportTestUtil.java

```diff
@@ -22,6 +24,53 @@ public static Repository createRepository() {
         return new MutableRepository(new MutableTrieCache(new MutableTrieImpl(null, new Trie())));
     }
 
+    public static PartialMerkleTree createValidPmtWithTransactions(List<Sha256Hash> hashesToAdd, NetworkParameters networkParameters) {
+        byte[] bits = new byte[1];
+        bits[0] = 0x3f;
+
+        return new PartialMerkleTree(networkParameters, bits, hashesToAdd, 1);
+    }
+
+    public static void recreateChainAtHeightWithStoredBlockWithPmtAtHeight(
+        BtcBlockStoreWithCache btcBlockStoreWithCache,
+        int chainHeight,
+        PartialMerkleTree partialMerkleTree,
+        int btcBlockWithPmtHeight,
+        NetworkParameters networkParameters
+    ) throws BlockStoreException {
+
+        // first create a block that has the wanted partial merkle tree
+        BtcBlock btcBlockWithPmt = createBtcBlockWithPmt(partialMerkleTree, networkParameters);
+        // store it on the chain at wanted height
+        StoredBlock storedBtcBlockThatHasTransactions = new StoredBlock(btcBlockWithPmt, BigInteger.ONE, btcBlockWithPmtHeight);
+        btcBlockStoreWithCache.put(storedBtcBlockThatHasTransactions);
+        btcBlockStoreWithCache.setMainChainBlock(btcBlockWithPmtHeight, btcBlockWithPmt.getHash());
```

Comment:
What is the main purpose of `setMainChainBlock`?

Reply:
for that block to be the main chain at that height. 
so [this](https://github.com/rsksmart/rskj/blob/recreate_valid_block_stored_chain/rskj-core/src/test/java/co/rsk/peg/BridgeSupportTestUtil.java#L77) happens without mocking :)